### PR TITLE
Assignment5

### PR DESCRIPTION
**Pull request by mcstudor**

### CODE REVIEW

**Comment by CDobbins1** [comment]
Reviewing this.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 1]
```diff
@@ -23,9 +24,11 @@ public static void ConfigureServices(IServiceCollection services)
...
             services.AddScoped<IGiftService, GiftService>();
             services.AddScoped<IUserService, UserService>();
-            services.AddScoped<IGroupService, GroupService>();
-
-            services.AddDbContext<ApplicationDbContext>();
+            services.AddScoped<IGroupService, GroupService>();
+
+            services.AddDbContext<ApplicationDbContext>(options =>
+                options.EnableSensitiveDataLogging()
+                    .UseSqlite("Data Source=Blog.db"));
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: This works but this name doesn't match the project.

File: SecretSanta/src/SecretSanta.Business/Dto/Gift.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace SecretSanta.Business.Dto
+{
+    public class Gift : GiftInput, IEntity
+    {
+        public int Id { get; set; }
+    }
+
+    public class GiftInput
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Business/Dto/Gift.cs]: In general we would move this to its own file.

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/BaseApiControllerTests.cs [hunk 3]
```diff
@@ -29,141 +71,136 @@ public void Constructor_RequiresService()
...
+                .Where(p => p.GetAttribute<RequiredAttribute>() != null);
+            foreach (var property in properties)
             {
-                return Task.FromResult(Items.Remove(found));
+                TInputDto input = CreateInput();
+                property.SetValue(input, null!);
+                string jsonBody = JsonSerializer.Serialize(input);
+                using StringContent stringContent = new StringContent(jsonBody, Encoding.UTF8, "application/json");
+                HttpResponseMessage response = await Client.PostAsync(BaseUrl(), stringContent);
+                Assert.IsFalse(response.IsSuccessStatusCode);
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/Controllers/BaseApiControllerTests.cs]: Your unit test name says 400, but this assert only checks for success. If the service were to return a 500 response this test would still pass.

**Pull request closed without merging**